app/src/main/java/com/example/footballmanager/ui/MasterViewModel.kt

```diff
@@ -44,19 +44,24 @@ class MasterViewModel @Inject constructor(
     fun getFixturesData(date: String = LocalDate.now().toString(), ctx: Context) {
         viewModelScope.launch {
             try {
```

Comment:
you could use `runCatching` here and refactor this part of code eg.
`retrievingByDateState = runCatching {}.getOrElse { ... }`